File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    n = len(numbers)
    
    # 모든 number에 대해 부호 대입해 보기
    def backtrack(i, current_sum):
        nonlocal answer
        if i == n:
            if current_sum == target:
                answer += 1
            return
        
        backtrack(i + 1, current_sum + numbers[i])
        backtrack(i + 1, current_sum - numbers[i])
    
    backtrack(0, 0)
                
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

int answer = 0;

void backtrack(int start, int cnt, const vector<int>& numbers, int target) {
    if (start == numbers.size()) {
        if (cnt == target) {
            answer++;
        }
        return;
    }

    backtrack(start + 1, cnt + numbers[start], numbers, target);
    backtrack(start + 1, cnt - numbers[start], numbers, target);
}

int solution(vector<int> numbers, int target) {

    backtrack(0, 0, numbers, target);
    
    return answer;
}